### ISSUE TITLE
Automation process does not work correctly

### ISSUE THREAD

**Issue by erokoder** [nfarina/homebridge#2177]
I do not think it's causing problems because homebridge but not 100% sure. The problem occurs during the automation process.

Automated Setup is: When basement motion sensor is activated, turn lights on, also similar automation.
Automation most often works correctly, but sometimes not.  I look into my  HOME app and I can saw that sensor is activated but lights don not turn on. Sometimes I know the problem is communication and HOME accessories didn't synchronize.  But in this case I check in real time when accessory synchronize, and I saw that sensor activated but the light does not turn on.

Does anybody met with this problem and what might be a problem?

**Comment by NorthernMan54** [comment]
If you run homebridge in DEBUG mode you can collect a lot of information about what is being communicated between your home hub and homebridge and can see what is happening and not happening.

Is your motion sensor and lights coming from homebridge or are they native HomeKit?

**Comment by erokoder** [comment]
@NorthernMan54 
I didnt run in DEBUG mode, but I will try.

Motion Sensor - Sonoff 433Mhz, my custom plugin.
Light - Sonoff Basic, also custom plugin.

When I press Light I dont have problem. Also when I move in range motion sensor. I can saw sensor activated. 
So I exclude problem in my custom hombridge plugin, also I exclude homebridge problem.
That automation for some reason does not pass in cca 3%. 97% time work correctly.

**Comment by NorthernMan54** [comment]
If you review the DEBUG log carefully, you should see your automation hub ( Apple TV, HomePod or IPad ) register for events from your motion sensor, the actual motion event from the sensor being sent to the automation hub, and the light control message being sent to homebridge.  All 3 messages need to happen for the light to turn on.

When you restart homebridge, the automation hub should register for events shortly afterwards.

**Comment by ronzelver** [comment]
It might be related to the fact that the Homekit hub (Apple TV) sometimes detaches from Homebridge resulting in some automation not working correctly or at all.
See also https://github.com/nfarina/homebridge/issues/2040#issuecomment-479052065

**Comment by erokoder** [comment]
@ronzelver 

My Setup:
Homebridge on Rpi (Static IP)
iPAD Home HUB (Static IP)
and three iphone device.

I don't have any problem when I turn on/off/close/open any device. Also, I don't have a problem with the remote. 
Sometimes(10% cases) my automation didn't work (Motion Sensor -> Turn on lights).
(I hold the iPad in my hand and I saw that homebridge connected, and saw motion activating but sometimes lights not want to turn on. In same time, I can turn light on/off.

Do you think the problem is a ipad for hub and wifi network, maybe it's a better choice for apple tv and ethernet?

I will upload video to see it.

**Comment by erokoder** [comment]
@NorthernMan54 

In situation when automation dont working I have only this message:
 EventedHTTPServer [::ffff:192.168.2.245] Sending HTTP event '6.10' with data: {"characteristics":[{"aid":6,"iid":10,"value":true}]} +812ms
  EventedHTTPServer [::ffff:192.168.2.240] Sending HTTP event '6.10' with data: {"characteristics":[{"aid":6,"iid":10,"value":true}]} +2ms

**Comment by NorthernMan54** [comment]
@erokoder you can look at that as 'device' 192.168.2.245 and 192.168.2.240 was told that accessory 6 characteristic 10 was set to true, what is missing is the turn on command afterwards.  It should come from your iPAD

Are you clearing the Motion detected status after a period of time?

But if you watch the commentary here, I think iPad's as automation hubs have been problematic for people and once they switched to apple tv's things improved.

**Comment by NorthernMan54** [comment]
Another option to try is thing node-red based automation engine for homebridge I'm working on

https://github.com/NorthernMan54/HAP-NodeRed

Still beta, but been using it for things that HomeKit automation can't do

**Comment by erokoder** [comment]
@NorthernMan54 @ronzelver 

Also one more example:

Mz Hallway has two motion detection: Entrance and Room and two lights (Room and Entrance).

Automation is: When any motion activates (Entrance or Room) turn both lights.

Event:
1. Room Activate (Both lights on).
2. I turn off Room Lights and again activate Room motion.
3. Automation doesn't turn on Room Lights.
4. I turn off Entrance Lights and again activate Room motion.
5. Automation doesn't turn on any lights.
6. I walk to entrance motion and activating.
7. Both Lights turn on.

Crazy :D

Link: https://www.youtube.com/watch?v=4BBU8uz0k8w

**Comment by erokoder** [comment]
> @erokoder you can look at that as 'device' 192.168.2.245 and 192.168.2.240 was told that accessory 6 characteristic 10 was set to true, what is missing is the turn on command afterwards. It should come from your iPAD
> 
> Are you clearing the Motion detected status after a period of time?
> 
> But if you watch the commentary here, I think iPad's as automation hubs have been problematic for people and once they switched to apple tv's things improved.

Yes I remove Motion detected status (You can see in another video)

**Comment by caboj** [comment]
I'm also having similar issues with the Xiaomi sensors, using the homebridge-mi-aqara plugin. Up until recently, things worked flawlessly.

Now, sensors will change and should trigger automations (ie. open door, turn on light), and I can see the sensor updating in Home app, but it doesn't trigger the automation.

Other things, like motion detectors and buttons (also from the homebridge-mi-aqara plugin) work sporadically, but the door sensors never work to trigger automations anymore.

Nothing was added or changed that caused the issues, but I've tried updating some things since then, but nothing works!
